### ISSUE TITLE
Wrong error reported for non-existent server

### ISSUE THREAD

**Issue by msorens** [chef/automate#3297]
## Describe the bug
Attempting to create an organization for a chef server that does not exist reports an incorrect error.
```
"error": "no org found with id \"foo\"",
```

## To Reproduce

Attempt to create an organization using a non-existent--but well-formed GUID--as a server ID:
```
$ curl -sSkH "api-token: $TOK" -X POST \
-d '{"name": "foo", "admin_user": "ADMIN_USER", "admin_key": "KEY"}' \
https://a2-dev.test/api/v0/infra/servers/b900bfa6-c699-4b34-bb4e-146901de1164/orgs?pretty
{
  "error": "no org found with id \"foo\"",
  "code": 5,
  "message": "no org found with id \"foo\"",
  "details": [
  ]
}
```
Two problems:
1. "foo" is the name, not the id
2. it should say "server not found" rather than "no org found" and include the suspect ID, for example:
```
"error": "b900bfa6-c699-4b34-bb4e-146901de1164: server not found"
```

## Expected behavior
The error should indicate the server does not exist.

**Comment by vsingh-msys** [comment]
This should definitely a bug, thanks for reporting it.

**Comment by vsingh-msys** [comment]
Fixes #3297 vai #3298
